src/main/java/eu/dissco/backend/repository/DigitalMediaObjectRepository.java

```diff
@@ -29,118 +27,48 @@ public class DigitalMediaObjectRepository {
 
   public DigitalMediaObject getLatestDigitalMediaById(String id) {
     return context.select(NEW_DIGITAL_MEDIA_OBJECT.asterisk())
-        .distinctOn(NEW_DIGITAL_MEDIA_OBJECT.ID).from(NEW_DIGITAL_MEDIA_OBJECT)
+        .from(NEW_DIGITAL_MEDIA_OBJECT)
         .where(NEW_DIGITAL_MEDIA_OBJECT.ID.eq(id))
-        .orderBy(NEW_DIGITAL_MEDIA_OBJECT.ID, NEW_DIGITAL_MEDIA_OBJECT.VERSION.desc())
         .fetchOne(this::mapToMultiMediaObject);
   }
 
   public JsonApiData getLatestDigitalMediaObjectByIdJsonResponse(String id) {
-
-    Field<Integer> maxSpeciesVersion = DSL.max(NEW_DIGITAL_SPECIMEN.VERSION)
-        .as("maxSpeciesVersion");
-
-    var specimenRecentVersions = context.select(NEW_DIGITAL_SPECIMEN.ID, maxSpeciesVersion)
-        .from(NEW_DIGITAL_SPECIMEN).groupBy(NEW_DIGITAL_SPECIMEN.ID).asTable();
-
     return context.select(NEW_DIGITAL_SPECIMEN.SPECIMEN_NAME, NEW_DIGITAL_SPECIMEN.VERSION,
-            NEW_DIGITAL_SPECIMEN.ID, NEW_DIGITAL_MEDIA_OBJECT.asterisk()).from(NEW_DIGITAL_SPECIMEN)
-        .join(specimenRecentVersions)
-        .on(NEW_DIGITAL_SPECIMEN.ID.eq(specimenRecentVersions.field(NEW_DIGITAL_SPECIMEN.ID)))
-        .and(NEW_DIGITAL_SPECIMEN.VERSION.eq(specimenRecentVersions.field(maxSpeciesVersion)))
+            NEW_DIGITAL_SPECIMEN.ID, NEW_DIGITAL_MEDIA_OBJECT.asterisk())
+        .from(NEW_DIGITAL_SPECIMEN)
         .join(NEW_DIGITAL_MEDIA_OBJECT)
         .on(NEW_DIGITAL_SPECIMEN.ID.eq(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID))
         .where(NEW_DIGITAL_MEDIA_OBJECT.ID.eq(id))
-        .orderBy(NEW_DIGITAL_MEDIA_OBJECT.ID, NEW_DIGITAL_MEDIA_OBJECT.VERSION.desc()).limit(1)
         .fetchOne(this::mapToJsonApiData);
   }
 
-  public List<JsonApiData> getAnnotationsOnDigitalMediaObject(String mediaId, int pageNumber, int pageSize){
-    var offset = 0;
-    if (pageNumber > 1) {
-      offset = offset + (pageSize * (pageNumber - 1));
-    }
-
-    return context.select(NEW_ANNOTATION.asterisk())
-        .from(NEW_ANNOTATION)
-        .where(NEW_ANNOTATION.TARGET_ID.eq(mediaId))
-        .offset(offset)
-        .limit(pageSize)
-        .fetch(this::mapAnnotationToJsonApiData);
-  }
-
-
   public List<DigitalMediaObject> getDigitalMediaForSpecimen(String id) {
     return context.select(NEW_DIGITAL_MEDIA_OBJECT.asterisk())
```

Comment:
I think this function and the previous functions' names should be swapped. This function gets latest digital media objects, the previous function gets media for a given specimen.

Reply:
Hmm no I don't think so, checked and this method uses the
`.where(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID.eq(id))` (line 48)
So it gives back all digitalMedia for a specific specimen, think the name is correct.